settings-patch.php

```diff
@@ -0,0 +1,311 @@
+				),
+				'blacklist_keys' => array(
+					'show_in_rest' => true,
+					'type' => 'string',
+					'description' => __( 'When a comment contains any of these words in its content, name, URL, email, or IP address, it will be put in the trash. One word or IP address per line. It will match inside words, so “press” will match “WordPress”.', 'wp-graphql' )
+				),
+				'show_avatars' => array(
+					'show_in_rest' => true,
+					'type' => 'boolean',
+					'description' => __( 'Show Avatars', 'wp-graphql' )
+				),
+				'avatar_rating' => array(
+					'show_in_rest' => array(
+						'schema' => array(
+							'enum' => array( 'G', 'PG', 'R', 'X' ),
+						),
+					),
+					'type' => 'string',
+					'description' => __( 'Avatar content rating', 'wp-graphql' )
+				),
+				'avatar_default' => array(
+					'show_in_rest' => array(
+						'schema' => array(
+							'enum' => array(
+								'mystery',
+								'blank',
+								'gravatar_default',
+								'identicon',
+								'wavatar',
+								'monsterid',
+								'retro'
+							),
+						),
+					),
+					'type' => 'string',
+					'description' => __( 'For users without a custom avatar of their own, you can either display a generic logo or a generated one based on their email address', 'wp-graphql' )
+				),
+				'close_comments_for_old_posts' => array(
+					'show_in_rest' => true,
+					'type' => 'boolean',
+					'description' => __( 'Automatically close comments on articles older than a number of days', 'wp-graphql' )
+				),
+				'close_comments_days_old' => array(
+					'show_in_rest' => true,
+					'type' => 'integer',
+					'description' => __( 'The number of days required for automatically closing the comments on an article', 'wp-graphql' )
+				),
+				'thread_comments' => array(
+					'show_in_rest' => true,
+					'type' => 'boolean',
+					'description' => __( 'Enable threaded (nested) comments', 'wp-graphql' ),
+					'default' => false
+				),
+				'thread_comments_depth' => array(
+					'show_in_rest' => true,
+					'type' => 'integer',
+					'description' => __( 'Maximum level of comment thread depth', 'wp-graphql' )
+				),
+				'page_comments' => array(
+					'show_in_rest' => true,
+					'type' => 'boolean',
+					'description' => __( 'Break comments into pages', 'wp-graphql' ),
+					'default' => false
+				),
+				'comments_per_page' => array(
+					'show_in_rest' => true,
+					'type' => 'integer',
+					'description' => __( 'Number of top level comments per page', 'wp-graphql' )
+				),
+				'default_comments_page' => array(
+					'show_in_rest' => array(
+						'schema' => array(
+							'enum' => array( 'newest', 'oldest' ),
+						),
+					),
+					'type' => 'string',
+					'description' => __( 'Comment page displayed by default', 'wp-graphql' )
+				),
+				'comment_order' => array(
+					'show_in_rest' => array(
+						'schema' => array(
+							'enum' => array( 'asc', 'desc' ),
```

Comment:
so for fields like this that are exposed as an Enum in REST should _probably_ also be exposed in GraphQL as an enum. 

No reason this should be a String in GraphQL if we have the power to validate against a static enum already 🤔

Reply:
`register_setting` doesn't have any `enum` type to my knowledge. `string`, `integer`, and `boolean` are the only types as far as I know. I used [register_initial_setting](https://github.com/WordPress/WordPress/blob/54d43d88ea5895813fac6045d0f2bbc55d4d389e/wp-includes/option.php#L1887) as an example.